src/fromager/requirements_file.py

```diff
@@ -19,20 +19,15 @@ class RequirementType(StrEnum):
     BUILD_BACKEND = "build-backend"
     BUILD_SDIST = "build-sdist"
 
-    def __eq__(self, other: object) -> bool:
-        if isinstance(other, RequirementType):
-            if self.value == "build" or other.value == "build":
-                allowed_values = [
-                    "build-backend",
-                    "build-system",
-                    "build-sdist",
-                    "build",
-                ]
-                return self.value in allowed_values and other.value in allowed_values
-        return super.__eq__(self, other)
-
-    def __ne__(self, value: object) -> bool:
-        return not self == value
+    @property
+    def is_build_requirement(self) -> bool:
+        """Is requirement a build time requirement?"""
+        return self.value in {"build", "build-system", "build-backend", "build-sdist"}
```

Comment:
I think the 'build' type is just here so we can check if something is any of the other build types, right? Do we still need that with the API change you're making?

Reply:
@dhellmann I have removed `BUILD` enum member.